README.md

```diff
@@ -69,7 +69,7 @@ This repository contains:
 `  
     `cd MarsRover2020-firmware`
 
-    **Note:** The repository has a submodule. To update the submodule, use `git submodule update
+    **Note:** The repository has a submodule. To update the submodule, use `git submodule update --remote --merge`
```

Comment:
this merges the latest remote version of the submodule, which I think is more convenient. Without this, to get the latest version, ppl need to cd in the submodule and git pull

Reply:
`git submodule update` should checkout the version of the submodule that matches the hash in the git modules file. no need to cd into the submodule

Reply:
From https://git-scm.com/docs/git-submodule:
`Update the registered submodules to match what the superproject expects by cloning missing submodules, fetching missing commits in submodules and updating the working tree of the submodules.`

Reply:
`git submodule update --remote --merge` will actually not respect the pinned hash in the gitmodules. It will merge in changes to whatever branch the submodule was on (which it shouldn't be on in the first place. should be pinned to a commit).

Reply:
Why do we want it pinned to a commit and not a branch? The reason I made this change was bc of this scenario: master of the submodule is updated. Now, in the fw repo, issuing a git submodule update does not bring in the recently made changes. Instead, a git pull is needed inside the submodule.

Shouldnt the submodule always be on the latest version of master no matter what? why tie it to a commit and not the master branch.

Reply:
Someone is working on a branch and everything compiles. If it's not tied to a commit, ur update to the hw bridge can cause this person's pipelines to fail and local builds to fail as well. 

The only time when the submodule should move to a different commit should be when u explicitly implement the feature added. 

Ex. You add new can ids to bridge. On a fw branch, u update the commit of hwbridge and then implement the stuff using the new ids. Then u pr the fw change. 

You dont want people using the new bridge copy until that fw pr is merged. Suppose u introduce a breaking change in the hw bridge repo.

Reply:
ah i see now thx ur last point cleared it up